Hyperparameter_Tuning/Hyperparameter_Tuning_MIMIC.py

```diff
@@ -0,0 +1,271 @@
+#%% -------- Import Libraries -------- #
+
+# Standard imports
+from selectors import EpollSelector
+from tokenize import String
+import numpy as np
+import pandas as pd
+import torch
+
+# VAE is in other folder
+import sys
+
+sys.path.append("../")
+
+# Opacus support for differential privacy
+from opacus.utils.uniform_sampler import UniformWithReplacementSampler
+
+# For VAE dataset formatting
+from torch.utils.data import TensorDataset, DataLoader
+
+# VAE functions
+from VAE import Decoder, Encoder, VAE
+
+# For datetime columns we need a transformer
+from rdt.transformers import datetime
+
+# Utility file contains all functions required to run notebook
+from utils import (
+    set_seed,
+    mimic_pre_proc,
+    constraint_filtering,
+    plot_elbo,
+    plot_likelihood_breakdown,
+    plot_variable_distributions,
+    reverse_transformers,
+)
+from metrics import distribution_metrics
+
+import optuna
+import pickle
+
+set_seed(0)
+
+filepath = ".../Private MIMIC Data/table_one_synthvae.csv"
+
+# Load in the MIMIC dataset
+data_supp = pd.read_csv(filepath)
+
+# Save the original columns
+
+original_categorical_columns = [
+    "ETHNICITY",
+    "DISCHARGE_LOCATION",
+    "GENDER",
+    "FIRST_CAREUNIT",
+    "VALUEUOM",
+    "LABEL",
+]
+original_continuous_columns = ["Unnamed: 0", "ROW_ID", "SUBJECT_ID", "VALUE", "age"]
```

Comment:
`"Unnamed: 0"` is coming from `index=False` missing in the writing out of the pandas dataframe from the MIMIC prepro script and is thus not useful.